### PR TITLE
test: add shadow-1-T2 feature

### PR DESCRIPTION
**Pull request by jiangpta**
**Issue #, if available:**
n/a
**Description of changes:**
added scenario Shadow-1-T2 to test updating shadow of custom component
**Why is this change necessary:**
adding UAT tests for shadow manager 
**How was this change tested:**
tested on remote ec2 Linux instance
**Any additional information or context required to review the change:**
some previous code was touched to pass the PMD checking.

**Checklist:**
 - [ ] Updated the README if applicable
 - [ ] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] If your code makes a remote network call, it was tested with a proxy
 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java [hunk 1]
```diff
@@ -52,54 +49,70 @@ public class LocalDeploymentSteps {
...
+     * @param componentName         the name of the custom component
+     * @param configurationTable    the table which describes the configurations
+     * @throws InterruptedException InterruptedException could be throw out during the component deployment
+     * @throws IOException          IOException could be throw out during preparation of the CLI command
+     */
+    @When("I update the component {word} with configuration")
+    public void updateComponentWithConfiguration(final String componentName, final DataTable configurationTable)
+            throws InterruptedException, IOException {
+        List<Map<String, String>> configuration = configurationTable.asMaps(String.class, String.class);
+        updateComponent(componentName, configuration);
```
Comment (Nelsonochoam) [inline uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java:113]: Seems like the contents of this method can be placed in here. `updateComponent`  is only called here and it not doing anything special, so delegating here is not adding any extra value nor is it separating different concerns

File: uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java [hunk 2]
```diff
@@ -119,6 +132,23 @@ private CommandInput prepareCliDeploymentCommand(String componentName, String co
                 .build();
     }
 
+    private CommandInput prepareCliUpdateCommand(String componentName,
```
Comment (Nelsonochoam) [inline uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java:135]:
```suggestion
    private CommandInput getCliUpdateConfigurationCommand(String componentName,
```

File: uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java [hunk 3]
```diff
@@ -119,6 +132,23 @@ private CommandInput prepareCliDeploymentCommand(String componentName, String co
...
+                                                     List<Map<String, String>> configuration) throws IOException {
+        List<String> commandArgs = new ArrayList<>(Arrays.asList(
```
Comment (Nelsonochoam) [inline uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java:137]:
```suggestion
        List<String> commandArgs = Arrays.asList(
```

File: uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java [hunk 4]
```diff
@@ -162,24 +193,27 @@ private void installComponent(List<String> component, List<Map<String, String>>
         }
     }
 
-    private String getCliUpdateConfigArgs(String componentName, List<Map<String, String>> configuration) throws IOException {
+    private void updateComponent(String component, List<Map<String, String>> configuration)
+            throws InterruptedException, IOException {
+            String componentName = component;
```
Comment (Nelsonochoam) [inline uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java:198]: What value is this adding? You could replace the parameter name and avoid this reassignment

File: uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java [hunk 5]
```diff
@@ -119,6 +132,23 @@ private CommandInput prepareCliDeploymentCommand(String componentName, String co
                 .build();
     }
 
+    private CommandInput getCliUpdateConfigurationCommand(String componentName,
+                                                     List<Map<String, String>> configuration) throws IOException {
+        List<String> commandArgs = Arrays.asList(
+                "deployment",
+                "create",
+                "--merge " + componentName));
```
Comment (saranyailla) [inline uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java:140]: merge arg takes both component name-version. I don't see version here. 
But if you only want to update the configuration of the locally installed components, you don't need `--merge` arg (which is what you're doing here). 

You can reuse the `prepareCliDeploymentCommand` and check for version. If it is null, do not include `--merge` arg.

File: uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java (hunk 1 above)
Comment (saranyailla) [inline uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java:113]: Rename this to `updateComponentConfiguration`

Comment (jiangpta) [reply uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java:113]: done

File: uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java (hunk 4 above)
Comment (jiangpta) [reply uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java:198]: done

File: uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java (hunk 1 above)
Comment (jiangpta) [reply uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java:113]: done

**Comment by saranyailla** [comment]
Look good to me.

**Review by saranyailla — APPROVED** [review]

File: uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java (hunk 5 above)
Comment (jiangpta) [reply uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java]: done.

**Review by Nelsonochoam — APPROVED** [review]

**Pull request merged**